hayeon/programmers-1/다트-게임.js

```diff
@@ -0,0 +1,35 @@
+function solution(dartResult) {
+  const numArr = [0, 0, 0];
+  let point = -1;
+
+  for (let i = 0; i < dartResult.length; i++) {
+    if (Number.isInteger(dartResult[i] * 1)) {
+      point++;
+
+      if (dartResult[i] === "1" && dartResult[i + 1] === "0") {
+        numArr[point] += 10;
+        i++;
+      } else numArr[point] += dartResult[i] * 1;
+    } else {
+      if (dartResult[i] === "D") numArr[point] **= 2;
+      else if (dartResult[i] === "T") numArr[point] **= 3;
+      else if (dartResult[i] === "*") {
+        numArr[point] *= 2;
+        numArr[point - 1] *= 2;
+      } else if (dartResult[i] === "#") numArr[point] *= -1;
+    }
+  }
+  return numArr.reduce((a, b) => a + b);
+}
```

Comment:
```suggestion
function solution(dartResult) {
  const numArr = [0, 0, 0];
  let point = -1;

  for (let i = 0; i < dartResult.length; i++) {
    if (Number.isInteger(dartResult[i] * 1)) {
      point++;

      if (dartResult[i] === "1" && dartResult[i + 1] === "0") {
        numArr[point] += 10;
        i++;
      } else {
        numArr[point] += dartResult[i] * 1;
      }
    } else {
      if (dartResult[i] === "D") {
        numArr[point] **= 2;
      } else if (dartResult[i] === "T") {
        numArr[point] **= 3;
      } else if (dartResult[i] === "*") {
        numArr[point] *= 2;
        numArr[point - 1] *= 2;
      } else if (dartResult[i] === "#") {
        numArr[point] *= -1;
      }
    }
  }
  return numArr.reduce((a, b) => a + b);
}
```
- 풀이잘봤습니다 🙇‍♂️ 피드백 드리고 싶은 부분이 연재님과 유사해서 연재님 피드백 확인해주시길 바랍니다!!
- https://github.com/Team-Grace/coding-test/pull/67

Reply:
넵 감사합니다!!!😆